### PR TITLE
Add per-route traffic split to API

### PR DESCRIPTION
**Pull request by adleong**
This API change allows per-route traffic splits to be represented in the service profile proxy API.  

Signed-off-by: Alex Leong <alex@buoyant.io>

### CODE REVIEW

File: proto/destination.proto [hunk 1]
```diff
@@ -236,3 +242,8 @@ message HttpStatusRange {
   uint32 max = 2;
 }
 
+message Backend {
+  string service = 1;
```
Comment (olix0r) [inline proto/destination.proto]: What is stored in this value? Is it a kubernetes service name? is it a new authority value? If the latter, it should not be called service.

File: proto/destination.proto [hunk 2]
```diff
@@ -164,6 +164,12 @@ message Route {
   // outstanding request will be cancelled, a timeout error response will be
   // returned, and no more retries will be attempted.
   google.protobuf.Duration timeout = 5;
+
+  // If this list is non-empty, traffic to this service should instead be split
+  // between the backends in this list.  Each backend should receive a portion
+  // of the traffic proportional to its weight.  If this list is empty, traffic
+  // should be sent to this service as normal.
+  repeated Backend traffic_split = 6;
```
Comment (olix0r) [inline proto/destination.proto]: I'm not a fan of the word "backend" here-- it's neither specific nor clear. and the phrase "traffic split", while it reflects the SMI naming, introduces a new term, traffic, into this API.

**Comment by adleong** [comment]
I'm torn about if the traffic split should be specified at the route level or at the service profile level.  Putting it at the route level as I have it here is attractive because it gives more flexibility in the ability to do per-route splits.  However, without the ability to specify traffic splits on the "default" route, we can't specify a traffic split for the entire service.

Perhaps a better approach would be to move traffic split up to the service level.  In the future if we'd like to support per-route splits, we can expand the API to allow per-route traffic split overrides which would take precedence over the service level traffic split.

File: proto/destination.proto (hunk 1 above)
Comment (olix0r) [reply proto/destination.proto]: Similarly, the semantics of weight values should be documented.

**Pull request merged**